File: easy/LeafCountInBinaryTree.cs
namespace Easy;

using DataStructures;

public class LeafCountInBinaryTree
{
    public int Solve(TreeNode? root)
    {
        if (root == null)
            return 0;
        if (root.Left == null && root.Right == null)
            return 1;
        return this.Solve(root.Left) + this.Solve(root.Right);
    }
}

File: easy/leaf_count_in_binary_tree.go
package easy

func LeafCountInBinaryTree(root *TreeNode) int {
	if root == nil {
		return 0
	}
	if root.Left == nil && root.Right == nil {
		return 1
	}
	return LeafCountInBinaryTree(root.Left) + LeafCountInBinaryTree(root.Right)
}
